service/service.go

```diff
@@ -0,0 +1,190 @@
+package service
+
+import (
+	"context"
+	"github.com/ONSdigital/dp-search-api/api"
+	"github.com/ONSdigital/dp-search-api/elasticsearch"
+	"github.com/ONSdigital/dp-search-api/query"
+	"github.com/ONSdigital/dp-search-api/transformer"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"os"
+
+	esauth "github.com/ONSdigital/dp-elasticsearch/v2/awsauth"
+	elastic "github.com/ONSdigital/dp-elasticsearch/v2/elasticsearch"
+	dphttp "github.com/ONSdigital/dp-net/http"
+	"github.com/ONSdigital/dp-search-api/config"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+type Service struct {
+	config              *config.Configuration
+	healthCheck         HealthChecker
+	api                 *api.SearchAPI
+	router              *mux.Router
+	server              HTTPServer
+	serviceList         *ExternalServiceList
+	esSigner            *esauth.Signer
+	elasticHTTPClient   dphttp.Clienter
+	queryBuilder        api.QueryBuilder
+	elasticSearchClient elasticsearch.Client
+	transformer         transformer.Transformer
+}
+
+//New creates a new service
+func New(cfg *config.Configuration, serviceList *ExternalServiceList) *Service {
+	svc := &Service{
+		config:      cfg,
+		serviceList: serviceList,
+	}
+	return svc
+}
+
+// SetServer sets the http server for a service
+func (svc *Service) SetServer(server HTTPServer) {
+	svc.server = server
+}
+
+// SetHealthCheck sets the healthchecker for a service
+func (svc *Service) SetHealthCheck(healthCheck HealthChecker) {
+	svc.healthCheck = healthCheck
+}
+
+// SetQueryBuilder sets the queryBuilder for a service
+func (svc *Service) SetQueryBuilder(queryBuilder api.QueryBuilder) {
+	svc.queryBuilder = queryBuilder
+}
+
+// SetEsSigner sets the AWS signer for a service
+func (svc *Service) SetEsSigner(esSigner *esauth.Signer) {
+	svc.esSigner = esSigner
+}
+
+//SetElasticHTTPClient sets the HTTP client for a service
```

Comment:
We shouldn't need both of these elasticsearch clients (line 63 and line 68) - we should be using one client and reusing it in this app.

My preference would be to get rid of `SetElasticHTTPClient` and keep `SetElasticSearchClient`

`SetElasticSearchClient` is actually using the elasticsearch package client we will want to use